terraform/modules/tf_rules_engine/iam.tf

```diff
@@ -24,14 +24,14 @@ data "aws_iam_policy_document" "read_threat_intel_table" {
 
 // Allow the Rules Engine to read the rules table
 resource "aws_iam_role_policy" "read_rules_table" {
-  count  = var.rules_table_arn == "" ? 0 : 1
+  count  = var.enable_rule_staging ? 1 : 0
```

Comment:
We probably don't need to define a new flag.  `var.rules_table_arn` is one of the outputs defined in `tf_globals` module and it is default value is empty string `""`
https://github.com/airbnb/streamalert/blob/ff3aa2183c88e6907a794919aeae573d65eee433/terraform/modules/tf_globals/output.tf#L1-L3  

Then this variable is referenced in rules_engine tf code here
https://github.com/airbnb/streamalert/blob/ff3aa2183c88e6907a794919aeae573d65eee433/streamalert_cli/terraform/rules_engine.py#L44

Let me see can I find the root cause.